src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -189,11 +189,11 @@ public static int compareToCI_UTF16(byte[] value, byte[] other) {
     }
 
     public static int hashCode(byte[] value) {
-        int h = 0;
-        for (byte v : value) {
-            h = 31 * h + (v & 0xff);
-        }
-        return h;
+        return switch (value.length) {
+            case 0 -> 0;
+            case 1 -> value[0];
```

Comment:
shouldn't that be: 
```
    case 1 -> value[0] & 0xff;
```

Reply:
Yes, good catch. I'll add a test case for negative latin1 bytes, too.